Sources/PackagePlugin/PluginContextDeserializer.swift

```diff
@@ -53,8 +53,8 @@ internal struct PluginContextDeserializer {
     /// Returns the `Target` that corresponds to the given ID (a small integer),
     /// or throws an error if the ID is invalid. The target is deserialized on-
     /// demand if it hasn't already been deserialized.
-    mutating func target(for id: WireInput.Target.Id) throws -> Target {
-        if let target = targetsById[id] { return target }
+    mutating func target(for id: WireInput.Target.Id, pluginGeneratedSources: [URL] = [], pluginGeneratedResources: [URL] = []) throws -> Target {
+        if let target = targetsById[id], target.sourceModule?.pluginGeneratedSources.count == pluginGeneratedSources.count, target.sourceModule?.pluginGeneratedResources.count == pluginGeneratedResources.count { return target }
```

Comment:
This seems a bit ugly.

Reply:
logic LGTM, but I see a usual formatting nit 😅